qiskit/pulse/commands/acquire.py

```diff
@@ -66,6 +59,14 @@ def __init__(self, duration: int, discriminator: Optional[Discriminator] = None,
         else:
             self._kernel = None
 
+        if discriminator:
```

Comment:
Equivalently:
```
        if discriminator and not isinstance(discriminator, Discriminator):
            raise PulseError('Invalid discriminator object is specified.')
        self._discriminator = discriminator
```
If you really want to handle the case where a user supplies `0` or `[]` or `False` as the value of `discriminator`, you can say `self._discriminator = discriminator or None`

Reply:
I see this is just being moved, but might as well clean it up since it's in the diff :)

Reply:
Fixed in f2035849